include/seqan3/utility/container/aligned_allocator.hpp

```diff
@@ -18,6 +18,11 @@
 
 #include <seqan3/core/platform.hpp>
 
+// __cpp_aligned_new is a C++17 feature that we use in this allocator and we require it.
+#if __cpp_aligned_new < 201606
+#    pragma GCC warning "Non-C++17 compliant compiler! Please open an issue with your compiler and platform!"
+#endif // __cpp_aligned_new < 201606
```

Comment:
Wouldn't this be something that goes into `platform.hpp`? Though I guess it would also prohibit you from using the rest of the library.

Reply:
Yeah, this should always be true, but on the other side I don't want this to be an annoyance for people who don't use this header in the case that it gives a warning. I don't have any strong feelings about this.